stackslib/src/net/api/setloglevel.rs.bak

```diff
@@ -0,0 +1,164 @@
+#[derive(Clone)]
+pub struct RPCSetLogLevelRequestHandler {
+    pub loglevel: Option<slog::Level>;
+    pub password: Option<String>,
+}
+
+impl RPCPostTransactionRequestHandler {
+    pub fn new(password: Option<String>) -> Self {
+        Self {
+            level: None,
+            password
+        }
+    }
+}
+
+/// Decode the HTTP request
+impl HttpRequest for RPCPostTransactionRequestHandler {
+    fn verb(&self) -> &'static str {
+        "POST"
+    }
+
+    fn path_regex(&self) -> Regex {
+        Regex::new(r#"^/v3/node/loglevel$"#).unwrap()
+    }
+
+    fn metrics_identifier(&self) -> &str {
+        "/v3/node/loglevel"
+    }
+
+    /// Try to decode this request.
+    /// There's nothing to load here, so just make sure the request is well-formed.
+    fn try_parse_request(
+        &mut self,
+        preamble: &HttpRequestPreamble,
+        _captures: &Captures,
+        query: Option<&str>,
+        body: &[u8],
+    ) -> Result<HttpRequestContents, Error> {
+        // If no authorization is set, then the block proposal endpoint is not enabled
+        let Some(password) = &self.auth else {
+            return Err(Error::Http(400, "Bad Request.".into()));
+        };
+        let Some(auth_header) = preamble.headers.get("authorization") else {
+            return Err(Error::Http(401, "Unauthorized".into()));
+        };
+        if auth_header != password {
+            return Err(Error::Http(401, "Unauthorized".into()));
+        }
+
+        if preamble.get_content_length() == 0 {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expected non-zero-length body for PostTransaction"
+                    .to_string(),
+            ));
+        }
+
+        if preamble.get_content_length() > MAX_PAYLOAD_LEN {
+            return Err(Error::DecodeError(
+                "Invalid Http request: PostTransaction body is too big".to_string(),
+            ));
+        }
+        
+        match preamble.content_type {
+            None => {
+                return Err(Error::DecodeError(
+                    "Missing Content-Type for transaction".to_string(),
+                ));
+            }
+            Some(HttpContentType::Text) => {
+                // TODO
+            
+            }
+            _ => {
+                return Err(Error::DecodeError(
+                    "Wrong Content-Type for loglevel; expected tex/plain".to_string(),
+                ));
+            }
+        }
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCSetLogLevelRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.loglevel = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+
+        // TODO
```

Comment:
Is this intended to be included yet?

Reply:
That whole file shouldn't be there. I'll remove it.